complete/_rg

```diff
@@ -173,6 +173,7 @@ _rg() {
     + '(zip)' # Compressed-file options
     {-z,--search-zip}'[search in compressed files]'
     $no"--no-search-zip[don't search in compressed files]"
+    {-P,--preprocessor}'[search files needing preprocessing]'
```

Comment:
If you're going to put these together, i would suggest renaming the group and changing the comment, since it's no longer accurate. Maybe `(preprocessor-zip) # File-preprocessing options` or something, to match the way `pretty-vimgrep` is done

Also, the spec isn't right. It should be something like:

```
{-P+,--preprocessor=}'[specify preprocessor utility]:preprocessor utility:_command_names -e'
```

(`_command_names -e` will prefer `PATH` executables, but also do the right thing if you try to complete an actual file path)